acapy_agent/core/plugin_registry.py

```diff
@@ -216,134 +253,183 @@ async def load_protocol_version(
-                        )
+                        LOGGER.trace("Loading routes from: %s", version_path)
+                        mod = ClassLoader.load_module(version_path)
                     except ModuleLoadError as e:
-                        LOGGER.error("Error loading admin routes: %s", e)
+                        LOGGER.error(
+                            "Error loading admin routes from %s: %s", version_path, e
+                        )
                         continue
-                    if mod and hasattr(mod, "register"):
-                        await mod.register(app)
             else:
                 # Load plugin routes that aren't in a versioned package.
+                routes_path = f"{plugin_name}.routes"
                 try:
-                    mod = ClassLoader.load_module(f"{plugin.__name__}.routes")
+                    LOGGER.trace("Loading non-versioned routes from: %s", routes_path)
+                    mod = ClassLoader.load_module(routes_path)
                 except ModuleLoadError as e:
-                    LOGGER.error("Error loading admin routes: %s", e)
+                    LOGGER.error("Error loading admin routes from %s: %s", routes_path, e)
                     continue
-                if mod and hasattr(mod, "register"):
-                    await mod.register(app)
 
-    def register_protocol_events(self, context: InjectionContext):
+            if mod and hasattr(mod, "register"):
+                LOGGER.trace("Registering routes for: %s", plugin_name)
+                await mod.register(app)
+
+    def register_protocol_events(self, context: InjectionContext) -> None:
         """Call route register_events methods on the current context."""
+        LOGGER.trace("Registering protocol events for %d plugins", len(self._plugins))
+
         event_bus = context.inject_or(EventBus)
         if not event_bus:
             LOGGER.error("No event bus in context")
             return
+
         for plugin in self._plugins.values():
-            definition = ClassLoader.load_module("definition", plugin.__name__)
+            plugin_name = plugin.__name__
+            LOGGER.trace("Processing events for plugin: %s", plugin_name)
+            mod = None
+            definition = ClassLoader.load_module("definition", plugin_name)
             if definition:
                 # Load plugin routes that are in a versioned package.
+                LOGGER.trace("Loading versioned events for: %s", plugin_name)
                 for plugin_version in definition.versions:
+                    version_path = f"{plugin_name}.{plugin_version['path']}.routes"
                     try:
-                        mod = ClassLoader.load_module(
-                            f"{plugin.__name__}.{plugin_version['path']}.routes"
-                        )
+                        LOGGER.trace("Loading events from: %s", version_path)
+                        mod = ClassLoader.load_module(version_path)
```

Comment:
Similar problem; we're overwriting `mod` with each version. Only the last mod's events are actually registered.